flycheck-haskell.el

```diff
@@ -251,13 +251,21 @@ buffer."
   "Process the a Cabal CONFIG."
   (let-alist config
     (setq-local flycheck-ghc-search-path
-                (append .build-directories .source-directories
+                (append (if (string-equal (car flycheck-haskell-runghc-command)
+                                          "stack")
```

Comment:
I am sorry, but I think that this is nonsense.  `flycheck-haskell-runghc-command` just tells you whether `stack` is installed, but that doesn't mean anything with regards to whether the current project uses Stack or Cabal.  

This codes makes Flycheck look for Stack directories even in projects that don't even have a `stack.yml` somewhere.  That's obviously not going to work.

I'm still not sure why exactly we need to make this difference.  Why exactly can't we just add both directories to the search path?